src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -77,34 +72,29 @@ const MyDiagnoses = () => {
   return (
     <Background>
       <View style={styles.container}>
-        <ForceRerenderOnNavigation resetStateFunction={refetchDiagnoses} />
         <Header isScrolling={scrolling} />
-        {downloadingDiagnoses && <ActivityIndicator size='large' />}
-        {diagnoses &&
-          <FlatList
-            ref={flatListRef}
-            data={diagnoses}
-            showsVerticalScrollIndicator={false}
-            renderItem={({ item }) => item}
-            keyExtractor={item => String(diagnoses.indexOf(item))}
-            onRefresh={getRenderedDiagnoses}
-            refreshing={refreshing}
-            disableVirtualization={false}
-            onScrollBeginDrag={() => setScrolling(true)}
-            onScrollEndDrag={({ nativeEvent }) => {
-              const snapToTopIfScrollingNearTheFirstElement = () => {
-                if (nativeEvent.contentOffset.y < OFFSET_THRESHOLD_TO_SNAP_FLATLIST) {
-                  flatListRef.current.scrollToIndex({ animated: true, index: 0 })
-                  setScrolling(false)
-                } else {
-                  setScrolling(true)
-                }
+        <FlatList
+          ref={flatListRef}
+          data={diagnoses.current}
+          showsVerticalScrollIndicator={false}
+          renderItem={({ item }) => item.renderedDiagnose}
+          keyExtractor={item => item.id}
+          disableVirtualization={false}
+          onScrollBeginDrag={() => setScrolling(true)}
+          onScrollEndDrag={({ nativeEvent }) => {
+            const snapToTopIfScrollingNearTheFirstElement = () => {
+              if (nativeEvent.contentOffset.y < OFFSET_THRESHOLD_TO_SNAP_FLATLIST) {
+                setScrolling(false)
+                flatListRef.current.scrollToIndex({ animated: true, index: 0 })
+              } else {
+                setScrolling(true)
               }
+            }
```

Comment:
Why is this function defined if it's called immediately?

Reply:
If it's because of readability, it can be improved by using a function as a predicate of the `if` statement:
```javascript
              if (isScrollWithinSnapThreshold(nativeEvent)) {
                setScrolling(false)
                flatListRef.current.scrollToIndex({ animated: true, index: 0 })
              } else {
                setScrolling(true)
              }
```

Reply:
I tried not to change too much of what was already there, I think we can change that. @MPucheta  what do you think?

Reply:
The [**boy scout rule**](https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html) tells us to leave the place better than we found it. If you find code that could be written better and it's a minor change, then improve it.